### ISSUE TITLE
Unloaded event not called on Window when app closed

### ISSUE THREAD

**Issue by MartinZikmund** [dotnet/wpf#1442]
* .NET Core Version: (haven't tested yet, will do)
* Windows version: Windows 10 16299
* Does the bug reproduce also in WPF for .NET Framework 4.8?: Yes

 **Problem description:**

When window is closed with the close button in the title bar, the `Window.Unloaded` event is not called. This is inconsistent behavior.

Interestingly enough, it can be worked around by creating a new instance of window in the constructor (does not even need to be stored anywhere):

```
public partial class MainWindow : Window
{
    public MainWindow()
    {
        InitializeComponent();

        new Window(); //<-- this will make Unloaded Event to trigger in WPF
    }
}
```
 
 **Actual behavior:** 

`Window.Unloaded` does not fire
 
 **Expected behavior:**

`Window.Unloaded` should fire.
 
 **Minimal repro:**

```
public partial class MainWindow : Window
{
    public MainWindow()
    {
        InitializeComponent();
        this.Unloaded += Debug.WriteLine("Hit");
    }
}

```

**Comment by MartinZikmund** [comment]
I will try to repro in WPF Core when at my PC, in case this is fixed I will close the issue

**Comment by MartinZikmund** [comment]
I can confirm the problem is present with 3.0.100-preview7-012821, updated the description.

**Comment by vatsan-madhavan** [comment]
Is this the last Window in your app, and this causes the app to shutdown ? 

See this. 

https://docs.microsoft.com/en-us/dotnet/api/system.windows.frameworkelement.unloaded?view=netframework-4.8

> Note that the Unloaded event is not raised after an application begins shutting down. Application shutdown occurs when the condition defined by the ShutdownMode property occurs. If you place cleanup code within a handler for the Unloaded event, such as for a Window or a UserControl, it may not be called as expected.

**Comment by MartinZikmund** [comment]
@vatsan-madhavan Yes, that is the case. The question is if this behavior should be kept or it should rather be changed in some future version for the sake of consistency. We were quite surprised as MvvmCross framework is using `Unloaded` to notify the view disappearing to the view model and we had the view model state saving logic in the method which handled this event. Technically this means the event itself is not usable, as you may never count on it being called (as theoretically any window can be the last).

**Comment by nzain** [comment]
If the event does not fire reliably, it should not be available to developers _without warning_. It could be marked as obsolete, or some Visual Studio code analysis could emit a warning.

**Comment by miloush** [comment]
@nzain there is a warning in the documentation, cited above

@MartinZikmund you can set the shutdown mode to `OnExplicitShutdown` and shutdown the application once you ensured all state is saved as needed
